src/main/java/li/cil/tis3d/common/module/DisplayModule.java

```diff
@@ -66,9 +73,12 @@ public static void tick() {
      * current state to newly connected/coming closer clients.
      */
     private final int[] image = new int[RESOLUTION * RESOLUTION];
-    private Object nativeImage;
     private boolean imageDirty = false;
 
+    private NativeImageBackedTexture backingTexture;
```

Comment:
Client class references break dedicated server. Includes this field and methods below used in rendering. Throwing `@Environment(EnvType.CLIENT)` at them should work.